src/AstTransforms.cpp

```diff
@@ -101,28 +101,36 @@ bool RemoveRelationCopiesTransformer::removeRelationCopies(AstTranslationUnit& t
             if (!isFact(*cl) && cl->getBodyLiterals().size() == 1u && bodyAtoms.size() == 1u) {
                 AstAtom* atom = bodyAtoms[0];
                 if (equal_targets(cl->getHead()->getArguments(), atom->getArguments())) {
-                    // we have a match but have to check that all arguments are either
-                    // variables or records containing variables
-                    bool onlyVars = true;
+                    // Requirements:
+                    // 1) (checked) It is a rule with exactly one body.
+                    // 3) (checked) The body consists of an atom.
+                    // 4) (checked) The atom's arguments must be identical to the rule's head.
+                    // 5) (pending) The rules's head must consist only of either:
+                    //  5a) Variables
+                    //  5b) Aggregates unpacked into variables
+                    // 6) (pending) Each variable must have a distinct name.
```

Comment:
Why are some of these marked as pending or checked? I don't mind the blurb about the requirements, though.

Reply:
`checked` have been verified above, `pending` are going to be checked below.